src/http-api/index.js

```diff
@@ -16,6 +18,10 @@ exports.start = (repo, callback) => {
     repo = undefined
   }
 
+  if (typeof repo === 'string') {
+    repo = new IPFSRepo(repo, {stores: fsbs})
+  }
```

Comment:
Since the `cli` is spawning a httpAPI by feeding a string and not a repo object, we need to convert it to become a repo object so that IPFS core can consume it (if we only had a type system!)